packages/eslint-plugin/configs/recommended-with-formatting.js

```diff
@@ -45,19 +40,4 @@ const config = {
 	},
 };
 
-if ( isPackageInstalled( 'jest' ) ) {
-	config.overrides = [
-		{
-			// Unit test files and their helpers only.
-			files: [ '**/@(test|__tests__)/**/*.js', '**/?(*.)test.js' ],
-			extends: [ require.resolve( './test-unit.js' ) ],
-		},
-		{
-			// End-to-end test files and their helpers only.
```

Comment:
Does the current setup for e2e tests conflicts in any way with Playwright? Maybe we could keep ESLint Plugin as it was and simply update `.test-e2e.js` to have rules for Playwright once the migration is done.

Reply:
The original setup conflicts with Playwright because it's targeting all 
`specs/**/*.js` and `*.spec.js` files anywhere in the project.

We could change the file/directory name or add overrides, that's how I did it before. However, IMO, the logic here is very problematic in the long term. Since that now Playwright will soon be considered the official e2e testing framework, it no longer makes sense to assume tests will all use jest + Puppeteer.

I think it's better to deprecate the automatic environment sniffing and instead require users to opt-in to individual presets.

Reply:
> Since that now Playwright will soon be considered the official e2e testing framework, it no longer makes sense to assume tests will all use jest + Puppeteer.

It isn't official until the migration is over so let's keep in mind that there are also other projects that use ESLint plugin and they make certain assumptions when opting for using Jest + Puppeteer. I guess we can leave without e2e rules overall in the plugin, but I'm a bit worried that we so easily change the setup without processing all cosequences.